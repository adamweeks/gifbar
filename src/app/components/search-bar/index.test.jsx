import React from 'react';
import renderer from 'react-test-renderer';

import SearchBar from '.';

describe(`SearchBar component`, () => {
  const doClear = jest.fn();
  const doExit = jest.fn();
  const doSearch = jest.fn();
  const onFocus = jest.fn();
  const component = renderer.create(
    <SearchBar
      doClear={doClear}
      doExit={doExit}
      doSearch={doSearch}
      onFocus={onFocus}
    />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});
