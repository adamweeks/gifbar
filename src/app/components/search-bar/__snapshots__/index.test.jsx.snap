// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders properly 1`] = `
<div
  className="searchBar"
>
  <form>
    <input
      id="searchBar"
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for gifs"
      type="text"
      value=""
    />
    <button
      className="cancel"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </form>
</div>
`;
